### PR TITLE
[SYCL][NFCI] Unify large-grf splitting with per-aspects split

### PR DESCRIPTION
**Pull request by AlexeySachkov**
The patch removes standalone splitter we had for `large-grf` and moves `large-grf` handling into per-aspect splitter.

The change is intended to be non-functional: at most it may affect the order and names of modules produced by `sycl-post-link`, but not their content.

### CODE REVIEW

File: llvm/tools/sycl-post-link/sycl-post-link.cpp [hunk 1]
```diff
@@ -806,108 +806,96 @@ processInputModule(std::unique_ptr<Module> M) {
...
-        string_vector Names;
-        M2.saveEntryPointNames(Names);
-        M2.cleanup(); // may remove some entry points, need to save/rebuild
-        M2.rebuildEntryPoints(Names);
-        MMs.clear();
-        MMs.emplace_back(std::move(M2));
-        DUMP_ENTRY_POINTS(MMs.back().entries(), MMs.back().Name.c_str(), 3);
-        Modified = true;
-      }
+    MDesc.fixupLinkageOfDirectInvokeSimdTargets();
```
Comment (AlexeySachkov) [inline llvm/tools/sycl-post-link/sycl-post-link.cpp:809]: Note for reviewers: this huge chunk of changes is essentially a removal of a `while` loop with shifting the body to the left for one indentation level

File: llvm/tools/sycl-post-link/ModuleSplitter.cpp [hunk 2]
```diff
@@ -886,8 +841,13 @@ getSplitterByOptionalFeatures(ModuleDesc &&MD,
   } else {
     Groups.reserve(PropertiesToFunctionsMap.size());
     for (auto &EPG : PropertiesToFunctionsMap) {
+      // Start with properties of a source module
+      auto Properties = MD.getEntryPointGroup().Props;
```
Comment (AlexeySachkov) [inline llvm/tools/sycl-post-link/ModuleSplitter.cpp:845]: Less `auto` should be used here

Comment (AlexeySachkov) [reply llvm/tools/sycl-post-link/ModuleSplitter.cpp]: 722fdd8b3bcb7efab54160a267b400b32134e022

File: llvm/test/tools/sycl-post-link/device-code-split/per-aspect-split-2.ll [hunk 3]
```diff
@@ -5,11 +5,11 @@
 ; RUN: FileCheck %s -input-file=%t.table --check-prefix CHECK-TABLE
```
Comment (sarnex) [inline llvm/test/tools/sycl-post-link/device-code-split/per-aspect-split-2.ll:5]: Should we add a test with aspects and large GRF to lock down the behavior?

**Review by sarnex — APPROVED** [review]
Nice, thanks for doing this

**Review by kbobrovs — APPROVED** [review]

File: llvm/test/tools/sycl-post-link/device-code-split/per-aspect-split-2.ll (hunk 3 above)
Comment (AlexeySachkov) [reply llvm/test/tools/sycl-post-link/device-code-split/per-aspect-split-2.ll:5]: The thing is that as we add more features, they will affect existing tests anyway, so I would prefer to leave this as-is to have an isolated test for a piece of the implementation

**Pull request merged**